source/Coop/Mod/CoopServer.cs

```diff
@@ -294,11 +344,14 @@ private void ReceiveClientPlayerPartyChanged(ConnectionBase connection, Packet p
             Persistence.MobilePartyEntityManager.GrantPartyControl(party, Persistence.ConnectedClients.Last());
         }
 
-        private void SendInitialWorldData(ConnectionServer connection)
+        private void SendHeroId(ConnectionBase connection, object obj)
         {
-            OnServerSendingWorldData?.Invoke();
-            connection.SendWorldData();
-            OnServerSentWorldData?.Invoke();
+            if (obj is PlayerHeroSerializer heroSerializer)
+            {
+                Hero hero = (Hero)heroSerializer.Deserialize();
+                Guid guid = CoopObjectManager.AddObject(hero);
```

Comment:
Adding the same hero again to the object store? We should try to make a debug UI for this tool so we can inspect it more easily.

Reply:
https://github.com/Bannerlord-Coop-Team/BannerlordCoop/issues/218